### ISSUE TITLE
Allow users to customize links

### ISSUE THREAD

**Issue by cammellos** [status-im/status-react#15469]
### Designs

https://www.figma.com/file/wA8Epdki2OWa8Vr067PCNQ/Composer-for-Mobile?node-id=2102-232933&t=tTYKjMpICnzwF5Zv-0

### Out of scope

- Enable link previews (we can assume for now that is always on)
- Multiple links (protocol should support them, but no need to implement it as of now from the sender point of view)

### Description

As a user types a link in the composer, we should unfurl it in status-go and return the information to the client, which will be given an option to remove the preview.

Once the user sends the link, it should pass those information to status-go, which will then create a message and send it over to the other end.

For the behavior of the composer, you can check with the design team, generally, what whatsapp does is probably the desired behavior, but double check with the design team.

This requires a change in the protocol.

Instead of sending the link (as it is now), we should attach the data about the link with the protobuf. 

Basically we should augment the protocol message with some link fields (a message could contain multiple) and then populate those.
